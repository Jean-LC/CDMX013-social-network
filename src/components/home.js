// eslint-disable-next-line import/no-unresolved
import { onAuthStateChanged } from 'https://www.gstatic.com/firebasejs/9.9.3/firebase-auth.js';
import { auth } from '../lib/auth.js';
import { onNavigate } from '../main.js';

export const home = () => {
  const sectionHome = document.createElement('section');
  sectionHome.setAttribute('class', 'section');

  // Header of home
  const headerHome = document.createElement('header');
  headerHome.setAttribute('class', 'header-home');
  // div for logo and home title
  const divLogo = document.createElement('div');
  divLogo.setAttribute('class', 'div-logo');
  const logoB = document.createElement('img');
  logoB.setAttribute('src', './img/Logo-B.png');
  logoB.setAttribute('class', 'logoB-header');
  const titleHeader = document.createElement('h2');
  titleHeader.setAttribute('class', 'title-header');
  titleHeader.textContent = 'HOME';

  divLogo.append(logoB, titleHeader);

  // div for create post
  const divCreatePost = document.createElement('div');
  divCreatePost.setAttribute('class', 'div-create-post');
  const imageCreatePost = document.createElement('img');
  imageCreatePost.setAttribute('src', './img/new-post.png');
  imageCreatePost.setAttribute('class', 'image-create-post');
  const titleCreatePost = document.createElement('p');
  titleCreatePost.setAttribute('class', 'title-create-post');
  titleCreatePost.textContent = 'share something';

  divCreatePost.append(imageCreatePost, titleCreatePost);

  headerHome.append(divLogo, divCreatePost);

  // section main
  const mainHome = document.createElement('main');
  mainHome.setAttribute('class', 'home-main');
  const sectionPosts = document.createElement('section');
  sectionPosts.setAttribute('class', 'section-posts');
  const navMenu = document.createElement('nav');
  navMenu.setAttribute('class', 'nav-menu');
  const indicatorDiv = document.createElement('div');
  indicatorDiv.setAttribute('class', 'indicator-div');
  const imageHomeNav = document.createElement('img');
  imageHomeNav.setAttribute('src', './img/home.png');
  imageHomeNav.setAttribute('class', 'image-home-nav');
  const dotIndicator = document.createElement('img');
  dotIndicator.setAttribute('src', './img/dot.png');
  dotIndicator.setAttribute('class', 'dot-indicator');
  const imageSearchNav = document.createElement('img');
  imageSearchNav.setAttribute('src', './img/search.png');
  imageSearchNav.setAttribute('class', 'image-search-nav');
  const imageUserNav = document.createElement('img');
  imageUserNav.setAttribute('src', './img/user.png');
  imageUserNav.setAttribute('class', 'image-profile-nav');

  indicatorDiv.append(imageHomeNav, dotIndicator);

  navMenu.append(indicatorDiv, imageSearchNav, imageUserNav);

  onAuthStateChanged(auth, (user) => {
    if (user) {
      console.log('user login');
    } else {
      onNavigate('/');
    }
  });

  imageUserNav.addEventListener('click', () => {
    onNavigate('/profile');
  });

  mainHome.append(sectionPosts, navMenu);

  sectionHome.append(headerHome, mainHome); 
  return sectionHome;
};
