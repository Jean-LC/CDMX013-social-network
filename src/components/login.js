/* eslint-disable max-len */
// eslint-disable-next-line import/no-unresolved
import { signInWithEmailAndPassword } from 'https://www.gstatic.com/firebasejs/9.9.3/firebase-auth.js';
import { onNavigate } from '../main.js';
import { auth } from '../lib/auth.js';

export const login = () => {
  // section login
  const sectionLogin = document.createElement('section');
  sectionLogin.setAttribute('id', 'login-page');

  // section header
  const headerLogin = document.createElement('header');
  headerLogin.setAttribute('class', 'login-header');

  // login label
  const Loginlabel = document.createElement('div');
  Loginlabel.setAttribute('class', 'header-label');
  Loginlabel.textContent = 'Log In';

  // logo of the login page
  const imgLogoLogin = document.createElement('img');
  imgLogoLogin.setAttribute('src', './img/Logo-BookNet.png');
  imgLogoLogin.setAttribute('class', 'logo-booknet');

  // appends img to header
  headerLogin.append(imgLogoLogin, Loginlabel);

  // main of login
  const mainLogin = document.createElement('main');
  mainLogin.setAttribute('class', 'login-main');

  // adds the buttons for login with email or google
  const loginButtonsDiv = document.createElement('div');
  loginButtonsDiv.setAttribute('class', 'register-buttons-div');
  const emailLoginButton = document.createElement('button');
  emailLoginButton.setAttribute('id', 'email-button');
  const googleLoginButton = document.createElement('button');
  googleLoginButton.setAttribute('id', 'google-button');

  // adds the back button
  const backButton = document.createElement('button');
  backButton.setAttribute('class', 'modal-button');
  backButton.setAttribute('id', 'backButtonLogin');
  backButton.textContent = ('Back');

  // adds the buttons for login to their div
  loginButtonsDiv.append(backButton, emailLoginButton, googleLoginButton);

  // adds dialog and form for a login modal
  const loginModal = document.createElement('dialog');
  loginModal.setAttribute('class', 'login-modal');
  const formLoginModal = document.createElement('form');
  formLoginModal.setAttribute('id', 'form-login-modal');
  const emailLoginInput = document.createElement('input');
  emailLoginInput.setAttribute('type', 'text');
  emailLoginInput.setAttribute('placeholder', 'email');
  emailLoginInput.setAttribute('class', 'login-input');
  const passLoginInput = document.createElement('input');
  passLoginInput.setAttribute('placeholder', 'password');
  passLoginInput.setAttribute('type', 'password');
  passLoginInput.setAttribute('class', 'login-input');
  const cancelLoginButton = document.createElement('button');
  cancelLoginButton.setAttribute('id', 'cancel-login-modal');
  cancelLoginButton.setAttribute('class', 'modal-button');
  cancelLoginButton.textContent = 'Back';
  const submitLoginButton = document.createElement('button');
  submitLoginButton.setAttribute('id', 'submit-login-modal');
  submitLoginButton.setAttribute('class', 'modal-button');
  submitLoginButton.textContent = 'Enter';

  // create paragrapho to show the error
  const paragraphError = document.createElement('p');
  paragraphError.setAttribute('class', 'error-paragraph');
  
  // appends the buttons and input text areas to form
  formLoginModal.append(emailLoginInput, passLoginInput, paragraphError, cancelLoginButton, submitLoginButton);

  // appends the form to the dialog tag
  loginModal.appendChild(formLoginModal);

  // appends the modal and buttons to main
  mainLogin.append(loginButtonsDiv, loginModal);

  // click listener for the back button to return to welcome
  backButton.addEventListener('click', () => {
    onNavigate('/');
  });

  // click listener for the emailLoginButton to show the loginModal
  emailLoginButton.addEventListener('click', () => {
    loginModal.showModal();
  });

  cancelLoginButton.addEventListener('click', (e) => {
    e.preventDefault();
    loginModal.close();
  });

  // click listener for the enter button to login with email and redirect to home
  submitLoginButton.addEventListener('click', (e) => {
    e.preventDefault();
    const loginEmail = emailLoginInput.value;
    const loginPassword = passLoginInput.value;
    signInWithEmailAndPassword(auth, loginEmail, loginPassword)
      .then(() => {
        onNavigate('/home');
      })
      .catch((error) => {
        const invalidEmail = 'Please check your information again';
        const wrongPassword = 'Wrong password, please try again';
        const userNotFound = 'This email is not registered, please Sign Up';
        const enterPassword = ' Please enter a password';
  
        if (error.code === 'auth/invalid-email') {
          paragraphError.innerText = invalidEmail;
        } else if (error.code === 'auth/wrong-password') {
          paragraphError.innerText = wrongPassword;
        } else if (error.code === 'auth/user-not-found') {
          paragraphError.innerText = userNotFound;
        } else if (error.code === 'uth/internal-error') {
          paragraphError.innerHTML = enterPassword;
        }
      });
  });

  sectionLogin.append(headerLogin, mainLogin);
  return sectionLogin;
};
