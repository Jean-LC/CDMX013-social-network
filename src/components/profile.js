import { onNavigate } from '../main.js';

export const profile = () => {
  const sectionProfile = document.createElement('section');
  
  // Header of Profile
  const headerProfile = document.createElement('header');
  headerProfile.setAttribute('class', 'header-profile');
  // div for logo and Profile title
  const divLogo = document.createElement('div');
  divLogo.setAttribute('class', 'div-logo');
  const logoB = document.createElement('img');
  logoB.setAttribute('src', './img/Logo-B.png');
  logoB.setAttribute('class', 'logoB-header');
  const titleHeader = document.createElement('h2');
  titleHeader.setAttribute('class', 'title-header');
  titleHeader.textContent = 'PROFILE';
  
  divLogo.append(logoB, titleHeader);
  
  // div for create post
  const divProfileBody = document.createElement('div');
  divProfileBody.setAttribute('class', 'div-profile-body');
  const imageSettingsProfile = document.createElement('img');
  imageSettingsProfile.setAttribute('src', './img/settings.png');
  imageSettingsProfile.setAttribute('class', 'image-settings-profile');
  const titleSettingsProfile = document.createElement('p');
  titleSettingsProfile.setAttribute('class', 'title-settings-profile');
  titleSettingsProfile.textContent = 'settings';

  // dropdown menu for setting
  const dialogSetting = document.createElement('dialog');
  dialogSetting.setAttribute('class', 'dialog-settings');
  // logout img
  const divLogoutImg = document.createElement('div');
  divLogoutImg.setAttribute('class', 'div-logout-img');
  const logoutImg = document.createElement('img');
  logoutImg.setAttribute('src', './img/logout.png');
  logoutImg.setAttribute('class', 'logout-img');
  const logoutTitle = document.createElement('p');
  logoutTitle.setAttribute('class', 'logout-title');
  logoutTitle.textContent = 'Log Out';
  // edit profile
  const divEditProfile = document.createElement('div');
  divEditProfile.setAttribute('class', 'div-edit-profile');
  const editImg = document.createElement('img');
  editImg.setAttribute('src', './img/edit.png');
  editImg.setAttribute('class', 'edit-img');
  const editTitle = document.createElement('p');
  editTitle.setAttribute('class', 'edit-title');
  editTitle.textContent = 'Edit';
  const closeSettings = document.createElement('p');
  closeSettings.setAttribute('class', 'close-settings');
  closeSettings.textContent = 'X';
  
  // appends items to div
  divLogoutImg.append(logoutImg, logoutTitle);
  divEditProfile.append(editImg, editTitle);
  // appends divs to div settings
  dialogSetting.append(divEditProfile, divLogoutImg, closeSettings);

  divProfileBody.addEventListener('click', () => {
    dialogSetting.show();
  });
  closeSettings.addEventListener('click', () => {
    dialogSetting.close();
  });
  
  divProfileBody.append(imageSettingsProfile, titleSettingsProfile);
  
  headerProfile.append(divLogo, divProfileBody);
  
  // section main
  const mainProfile = document.createElement('main');
  mainProfile.setAttribute('class', 'profile-main');

  // secction main of profile where the post from the user will be seen
  const sectionProfileMain = document.createElement('section');
  sectionProfileMain.setAttribute('class', 'section-profile-main');

  // nav bar to go to other pages
  const navMenu = document.createElement('nav');
  navMenu.setAttribute('class', 'nav-menu');

  // img to go to other pages
  const imageHomeNav = document.createElement('img');
  imageHomeNav.setAttribute('src', './img/home.png');
  imageHomeNav.setAttribute('class', 'image-home-nav');
  const imageSearchNav = document.createElement('img');
  imageSearchNav.setAttribute('src', './img/search.png');
  imageSearchNav.setAttribute('class', 'image-search-nav');
  const imageUserNav = document.createElement('img');
  imageUserNav.setAttribute('src', './img/user.png');
  imageUserNav.setAttribute('class', 'image-profile-nav');
  // indicator: where you at
  const indicatorDiv = document.createElement('div');
  indicatorDiv.setAttribute('class', 'indicator-div');
  const dotIndicator = document.createElement('img');
  dotIndicator.setAttribute('src', './img/dot.png');
  dotIndicator.setAttribute('class', 'dot-indicator');
  
  indicatorDiv.append(imageUserNav, dotIndicator);
  
  navMenu.append(imageHomeNav, imageSearchNav, indicatorDiv);
  
  mainProfile.append(dialogSetting, sectionProfileMain, navMenu);
  
  imageHomeNav.addEventListener('click', () => {
    onNavigate('/home');
  });
  
  sectionProfile.append(headerProfile, mainProfile); 
  return sectionProfile;
};
