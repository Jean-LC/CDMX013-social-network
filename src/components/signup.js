import { onNavigate } from '../main.js';

export const signup = () => {
  const sectionSignup = document.createElement('section');
  sectionSignup.setAttribute('id', 'signupPage');

  // section signup header
  const headerSignup = document.createElement('header');
  headerSignup.setAttribute('class', 'register-header');

  // login label
  const Signuplabel = document.createElement('div');
  Signuplabel.setAttribute('class', 'header-label');
  Signuplabel.textContent = 'Sign Up';

  // logo of the welcome page
  const imgLogoSignup = document.createElement('img');
  imgLogoSignup.setAttribute('src', './img/Logo-BookNet.png');
  imgLogoSignup.setAttribute('class', 'logo-booknet');

  // appends img to header
  headerSignup.append(imgLogoSignup, Signuplabel);

  // main of signup
  const mainSignup = document.createElement('main');
  mainSignup.setAttribute('class', 'register-main');

  // adds the buttons for signup with email or google
  const signupButtonsDiv = document.createElement('div');
  signupButtonsDiv.setAttribute('class', 'register-buttons-div');
  const emailSignupButton = document.createElement('button');
  emailSignupButton.setAttribute('id', 'email-button');
  const googleSignupButton = document.createElement('button');
  googleSignupButton.setAttribute('id', 'google-button');

  // adds the back button
  const backButton = document.createElement('button');
  backButton.setAttribute('class', 'modal-button');
  backButton.setAttribute('id', 'backButtonSignup');
  backButton.textContent = ('Back');

  // appends the signup div in main
  mainSignup.appendChild(signupButtonsDiv);

  // adds the buttons for signup to their div
  signupButtonsDiv.append(backButton, emailSignupButton, googleSignupButton);

  // adds dialog and form for a signup modal
  const signupModal = document.createElement('dialog');
  signupModal.setAttribute('class', 'signup-modal');
  const formSignupModal = document.createElement('form');
  formSignupModal.setAttribute('id', 'form-signup-modal');
  const emailSignupInput = document.createElement('input');
  emailSignupInput.setAttribute('type', 'text');
  emailSignupInput.setAttribute('placeholder', 'email');
  emailSignupInput.setAttribute('class', 'signup-input');
  const passSignupInput = document.createElement('input');
  passSignupInput.setAttribute('placeholder', 'password');
  passSignupInput.setAttribute('type', 'password');
  passSignupInput.setAttribute('class', 'signup-input');
  const cancelSignupButton = document.createElement('button');
  cancelSignupButton.setAttribute('class', 'modal-button');
  cancelSignupButton.setAttribute('id', 'cancel-signup-modal');
  cancelSignupButton.textContent = 'Back';
  const submitSignupButton = document.createElement('button');
  submitSignupButton.setAttribute('class', 'modal-button');
  submitSignupButton.setAttribute('id', 'submit-signup-modal');
  submitSignupButton.textContent = 'Enter';

  // appends the buttons and input text areas to form
  formSignupModal.append(emailSignupInput, passSignupInput, cancelSignupButton, submitSignupButton);

  // appends the form to the dialog tag
  signupModal.appendChild(formSignupModal);

  // appends the dialog to signup section
  mainSignup.appendChild(signupModal);

  // click listener  for the back button
  backButton.addEventListener('click', () => {
    onNavigate('/');
  });

  // click listener for the emailLoginButton to show the loginModal
  emailSignupButton.addEventListener('click', () => {
    signupModal.showModal();
  }); 

  cancelSignupButton.addEventListener('click', (e) => {
    e.preventDefault();
    signupModal.close();
  }); 

  sectionSignup.append(headerSignup, mainSignup);
  return sectionSignup;
};
