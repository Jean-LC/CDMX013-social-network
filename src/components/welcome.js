// import { onNavigate } from '../main.js';
import { googleButton } from './googleButton.js';

export const welcome = () => {
  // section welcome
  const sectionWelcome = document.createElement('section');
  sectionWelcome.setAttribute('id', 'register-page');

  // section header
  const headerWelcome = document.createElement('header');
  headerWelcome.setAttribute('class', 'register-header');

  // logo of the welcome page
  const imgLogoWelcome = document.createElement('img');
  imgLogoWelcome.setAttribute('src', './img/Logo-BookNet.png');
  imgLogoWelcome.setAttribute('class', 'logo-booknet');

  // appends img to header
  headerWelcome.appendChild(imgLogoWelcome);

  // main of welcome
  const mainWelcome = document.createElement('main');
  mainWelcome.setAttribute('class', 'register-main');

  // buttons for register and login
  const loginButton = document.createElement('button');
  loginButton.setAttribute('id', 'log-in');
  loginButton.setAttribute('class', 'register-button');
  loginButton.textContent = 'Log In';
  const signupButton = document.createElement('button');
  signupButton.setAttribute('id', 'sign-up');
  signupButton.setAttribute('class', 'register-button');
  signupButton.textContent = 'Sign up';

  // appends the buttons to the main tag
  mainWelcome.append(loginButton, signupButton);

  // adds the buttons for login with email or google
  const loginButtonsDiv = document.createElement('div');
  loginButtonsDiv.setAttribute('class', 'login-buttons-div');
  const emailLoginButton = document.createElement('button');
  emailLoginButton.setAttribute('id', 'email-login-button');
  emailLoginButton.textContent = 'Continue with email';
  const googleLoginButton = document.createElement('div');
  googleLoginButton.setAttribute('id', 'buttonDiv');

  // appends the googleButton module to the google login button
  googleLoginButton.append(googleButton());

  // adds the buttons for login to their div
  loginButtonsDiv.append(emailLoginButton, googleLoginButton);

  // adds dialog and form for a login modal
  const loginModal = document.createElement('dialog');
  loginModal.setAttribute('id', 'login-modal');
  const formLoginModal = document.createElement('form');
  formLoginModal.setAttribute('id', 'form-login-modal');
  const emailLoginInput = document.createElement('input');
  emailLoginInput.setAttribute('type', 'text');
  emailLoginInput.setAttribute('placeholder', 'email');
  emailLoginInput.setAttribute('class', 'login-input');
  const passLoginInput = document.createElement('input');
  passLoginInput.setAttribute('placeholder', 'password');
  passLoginInput.setAttribute('type', 'password');
  passLoginInput.setAttribute('class', 'login-input');
  const cancelLoginButton = document.createElement('button');
  cancelLoginButton.setAttribute('class', 'modal-login-button');
  cancelLoginButton.textContent = 'Back';
  const submitLoginButton = document.createElement('button');
  submitLoginButton.setAttribute('class', 'modal-login-button');
  submitLoginButton.textContent = 'Enter';
  
  // appends the buttons and input text areas to form
  formLoginModal.append(emailLoginInput, passLoginInput, cancelLoginButton, submitLoginButton);

  // appends the form to the dialog tag
  loginModal.appendChild(formLoginModal);

  // appends the dialog to section welcome
  sectionWelcome.appendChild(loginModal);

  // click listener for the emailLoginButton to show the loginModal
  emailLoginButton.addEventListener('click', () => {
    loginModal.showModal();
  });

  cancelLoginButton.addEventListener('click', (e) => {
    e.preventDefault();
    loginModal.close();
  });

  // appends the header and main to section
  sectionWelcome.append(headerWelcome, mainWelcome, loginButtonsDiv);
  return sectionWelcome;
};
