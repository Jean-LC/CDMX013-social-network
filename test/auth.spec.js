import { signup } from '../src/components/signup.js';
import { welcome } from '../src/components/welcome.js';
import { login } from '../src/components/login.js';
import { home } from '../src/components/home.js';
import { profile } from '../src/components/profile.js';

/** @jest-environment jsdom */

jest.mock('../src/main.js');
jest.mock('../src/lib/imports.js');

describe('welcome tests', () => {
  test('is welcome a function?', () => {
    expect(typeof welcome).toBe('function');
  });
  test('snapshot from welcome', () => {  
    expect(welcome()).toMatchSnapshot();
  });
  test('login button is not null', () => {
    const element = welcome();
    const loginButton = element.querySelector('#log-in');
    expect(loginButton).not.toBeNull();
  });
  test('signup button is not null', () => {
    const element = welcome();
    const signupButton = element.querySelector('#sign-up');
    expect(signupButton).not.toBeNull();
  });
});

describe('signup tests', () => {
  test('is signup a function?', () => {
    expect(typeof signup).toBe('function');
  });
  test('snapshot from signup', () => {  
    expect(signup()).toMatchSnapshot();
  });
  test('back button is not null', () => {
    const element = signup();
    const backButton = element.querySelector('.modal-button');
    expect(backButton).not.toBeNull();
  });
});

describe('login tests', () => {
  test('is login a function?', () => {
    expect(typeof login).toBe('function');
  });
  test('snapshot from signup', () => {  
    expect(login()).toMatchSnapshot();
  });
  test('back button is not null', () => {
    const element = login();
    const backButton = element.querySelector('.modal-button');
    expect(backButton).not.toBeNull();
  });
});

describe('home tests', () => {
  test('is home a function?', () => {
    expect(typeof home).toBe('function');
  });
  test('snapshot from home', () => {  
    expect(home()).toMatchSnapshot();
  });
  test('create post button is not null', () => {
    const element = home();
    const createPost = element.querySelector('.div-create-post');
    expect(createPost).not.toBeNull();
  });
  test('profile button is not null', () => {
    const element = home();
    const profileButton = element.querySelector('.image-profile-nav');
    expect(profileButton).not.toBeNull();
  });
});

describe('profile tests', () => {
  test('is profile a function?', () => {
    expect(typeof profile).toBe('function');
  });
  test('config button is not null', () => {
    const element = profile();
    const configButton = element.querySelector('.div-profile-body');
    expect(configButton).not.toBeNull();
  });
  test('home button is not null', () => {
    const element = profile();
    const homeButton = element.querySelector('.image-home-nav');
    expect(homeButton).not.toBeNull();
  });
});
