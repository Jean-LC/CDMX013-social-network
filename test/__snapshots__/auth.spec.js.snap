// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home tests snapshot from home 1`] = `
<section
  class="section"
>
  <header
    class="header-home"
  >
    <div
      class="div-logo"
    >
      <img
        class="logoB-header"
        src="./img/Logo-B.png"
      />
      <h2
        class="title-header"
      >
        HOME
      </h2>
    </div>
    <div
      class="div-create-post"
    >
      <img
        class="image-create-post"
        src="./img/new-post.png"
      />
      <p
        class="title-create-post"
      >
        share something
      </p>
    </div>
  </header>
  <main
    class="home-main"
  >
    <section
      class="section-posts"
    />
    <dialog
      class="new-post"
    >
      <div
        class="div-user-new-post"
      >
        <img
          class="user-image"
          src="./img/user-image.png"
        />
        <p
          class="email-user-new-post"
        />
      </div>
      <textarea
        class="input-new-post"
        cols="20"
        maxlength="200"
        placeholder="share something"
        rows="20"
      />
      <p
        class="count-paragraph"
      >
        0/200
      </p>
      <p
        class="error-post"
      />
      <div
        class="div-buttons-new-post"
      >
        <div
          class="cancel-post"
        >
          <img
            class="cancel-post-button"
            src="./img/cancel.png"
          />
          <p
            class="cancel-post-text"
          >
            cancel
          </p>
        </div>
        <div
          class="save-post-div"
        >
          <img
            class="save-post-button"
            src="./img/save-new.png"
          />
          <p
            class="save-post-text"
          >
            save
          </p>
        </div>
      </div>
    </dialog>
  </main>
  <nav
    class="nav-menu"
  >
    <div
      class="indicator-div"
    >
      <img
        class="image-home-nav"
        src="./img/home.png"
      />
      <img
        class="dot-indicator"
        src="./img/dot.png"
      />
    </div>
    <img
      class="image-profile-nav"
      src="./img/user.png"
    />
  </nav>
</section>
`;

exports[`login tests snapshot from signup 1`] = `
<section
  class="section"
>
  <div
    class="header-label"
  >
    Log In
  </div>
  <header
    class="login-header"
  >
    <img
      class="logo-booknet"
      src="./img/Logo-BookNet.png"
    />
  </header>
  <main
    class="login-main"
  >
    <div
      class="register-buttons-div"
    >
      <button
        class="modal-button"
        id="backButtonLogin"
      >
        Back
      </button>
      <button
        id="email-button"
      />
      <button
        id="google-button"
      />
    </div>
    <dialog
      class="login-modal"
    >
      <form
        id="form-login-modal"
      >
        <input
          class="login-input"
          placeholder="email"
          type="email"
        />
        <input
          class="login-input"
          placeholder="password"
          type="password"
        />
        <p
          class="error-paragraph"
        />
        <button
          class="modal-button"
          id="cancel-login-modal"
        >
          Back
        </button>
        <button
          class="modal-button"
          id="submit-login-modal"
        >
          Enter
        </button>
      </form>
    </dialog>
  </main>
</section>
`;

exports[`signup tests snapshot from signup 1`] = `
<section
  class="section"
>
  <div
    class="header-label"
  >
    Sign Up
  </div>
  <header
    class="register-header"
  >
    <img
      class="logo-booknet"
      src="./img/Logo-BookNet.png"
    />
  </header>
  <main
    class="register-main"
  >
    <div
      class="register-buttons-div"
    >
      <button
        class="modal-button"
        id="backButtonSignup"
      >
        Back
      </button>
      <p
        class="paragraph-terms"
      >
        By signing up, you agree to the Terms of Service and Privacy Policy, including Cookie Use.
      </p>
      <button
        id="email-button"
      />
      <button
        id="google-button"
      />
    </div>
    <dialog
      class="signup-modal"
    >
      <form
        id="form-signup-modal"
      >
        <input
          class="signup-input"
          placeholder="email"
          type="email"
        />
        <input
          class="signup-input"
          placeholder="password"
          type="password"
        />
        <p
          class="error-paragraph"
        />
        <button
          class="modal-button"
          id="cancel-signup-modal"
        >
          Back
        </button>
        <button
          class="modal-button"
          id="submit-signup-modal"
        >
          Enter
        </button>
      </form>
    </dialog>
  </main>
</section>
`;

exports[`welcome tests snapshot from welcome 1`] = `
<section
  class="section"
>
  <header
    class="welcome-header"
  >
    <img
      class="logo-booknet"
      src="./img/Logo-BookNet.png"
    />
  </header>
  <main
    class="register-main"
  >
    <button
      class="register-button"
      id="log-in"
    >
      Log In
    </button>
    <button
      class="register-button"
      id="sign-up"
    >
      Sign up
    </button>
  </main>
  <footer
    class="footer-welcome"
  >
    ©️2022 Jean y Lala
  </footer>
</section>
`;
